### PR TITLE
Allow uploading gd.games thumbnail directly

### PR DESCRIPTION
**Pull request by ClementPasteau**
Moved the Icons (PlatformSpecificAssets) to a tab of the Game Properties Dialog, but kept the menu item redirecting there for the moment. (I think should be removed after a bit)

https://github.com/user-attachments/assets/25f37919-6cd5-43ca-a284-9ea9423061f1

Game Dashboard flow:

https://github.com/user-attachments/assets/f88b7dfb-66e3-4038-a7b2-739891c1fd5b

Export flow:

https://github.com/user-attachments/assets/bb2b66d3-fb17-4f68-bd65-73b859d10c72

### CODE REVIEW

File: newIDE/app/src/GameDashboard/GameThumbnail.js [hunk 1]
```diff
@@ -1,54 +1,209 @@
...
+      console.error("Could't find selected file. Aborting thumbnail update.");
+      return;
+    }
+    const file = gamesPlatformThumbnailFileInputRef.current.files[0];
+    const fileType = file.type;
+    if (
+      !fileType ||
+      (fileType !== 'image/png' &&
+        fileType !== 'image/jpeg' &&
+        fileType !== 'image/webp')
```
Comment (AlexandreSi) [inline newIDE/app/src/GameDashboard/GameThumbnail.js:87]: Should we prevent webp format? It's not supported by every browser

**Review by AlexandreSi — APPROVED** [review]
LGTM, great GameThumbnail component

File: newIDE/app/src/GameDashboard/GameThumbnail.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/GameDashboard/GameThumbnail.js:87]: I think that's ok, it's here in Safari 16 (and Safari 15 on certain macos version), so overall it should be ok
image

File: newIDE/app/src/PlatformSpecificAssetsEditor/PlatformSpecificAssets.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        setIsLoading(false);
+      }, 200 /* Let a bit of time so that image files can be found */);
+    } catch (e) {
+      showAlert({
+        title: t`Some icons could not be generated.`,
+        message: t`An error occurred while generating some icons. Verify that the image is valid and try again.`,
+      });
+      setIsLoading(false);
+    } finally {
+      // Reset input value so that if the user selects the same file again,
```
Comment (4ian) [inline newIDE/app/src/PlatformSpecificAssetsEditor/PlatformSpecificAssets.js:182]: See if you can/should call [revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) here?

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/GameDashboard/GameDetails.js [hunk 3]
```diff
@@ -543,18 +552,21 @@ const GameDetails = ({
                     multiline
                     rows={5}
                   />
-                  <SelectField
-                    disabled
+                  <TextField
+                    value={
```
[the remark this answers is not in the document]
Comment (ClementPasteau) [reply newIDE/app/src/GameDashboard/GameDetails.js:556]: It was already a disabled SelectField! Which looked really weird, so using a textfield to make it more in line with the rest of that page

**Pull request merged**